docs/contributor/arch/008-telemetry-flow-healthiness-status-api.md

```diff
@@ -0,0 +1,102 @@
+
+* Using multiple condition types to represent various telemetry flow events (Throttling, Data Loss, High Buffer Utilization, etc.).
+* Using a single condition type (`TelemetryFlowHealthy`) with a reason field to denote diverse telemetry flow events.
+
+After careful consideration, we have opted for the single condition type `TelemetryFlowHealthy`, because it is easier to understand for the end user.
+Ultimately, the user's primary concern is understanding whether the telemetry flow is functioning correctly. In case of any issues, the user can do the following:
+
+* Troubleshoot the backend.
+* Investigate backend connectivity.
+* Reduce ingestion.
+* Manually scale out the gateway (as long as no autoscaling capability is provided).
+
+With a single condition type, we can assign the **reason** field a value that is most relevant for the user:
+```
+FullDataLoss > PartialDataLoss > HighBufferUtilization > GatewayThrottling > Healthy
+```
+
+The reasons are based on the following alert rules (an example for Metric Pipelines; the actual PromQL expressions must be defined later):
+| Alert Rule | Expression |
+| --- | --- |
+| GatewayExporterSentMetrics    | `sum(rate(otelcol_exporter_sent_metric_points{...}[5m])) > 0`           |
+| GatewayExporterDroppedMetrics  | `sum(rate(otelcol_exporter_send_failed_metric_points{...}[5m])) > 0`    |
+| GatewayReceiverRefusedMetrics  | `sum(rate(otelcol_receiver_refused_metric_points{...}[5m])) > 0`        |
+| GatewayExporterEnqueueFailed   | `sum(rate(otelcol_exporter_enqueue_failed_metric_points{...}[5m])) > 0` |
+| GatewayExporterQueueAlmostFull | `otelcol_exporter_queue_size / otelcol_exporter_queue_capacity > 0.8`   |                               |
+
+We could then map the alert rules to the reasons as follows:
+
+| Reason | Alert Rules |
+| --- | --- |
+| FullDataLoss           | **not** GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed) |
+| PartialDataLoss        | GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed)       |
+| HighBufferUtilization  | GatewayExporterQueueAlmostFull                                                                           |
+| GatewayThrottling      | GatewayReceiverRefusedMetrics                                                                            |
+| Healthy                | **not** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed **or** HighBufferUtilization **or** GatewayReceiverRefusedMetrics) |
+
+### Gateway/Agent Communication
+
+The mentioned events are about the Gateway Collector. Some issues might happen with the Agent, like:
+* Gateway can't be reached (network problem).
+* Gateway won't accept data (gateway throttling).
+
+In both cases, the Agent might drop data after too many retries. But because users usually can't do much about these issues, we won't include them in the `TelemetryFlowHealthy` condition.
```

Comment:
Does this mean the user will experience issues in the Telemetry flow, even though `TelemetryFlowHealthy` shows it's all fine?

Reply:
This feels kind of misleading :D